synapse/module_api/__init__.py

```diff
@@ -1274,6 +1274,24 @@ async def check_username(self, username: str) -> None:
         """
         await self._registration_handler.check_username(username)
 
+    async def store_remote_3pid_association(
```

Comment:
From the eyes of a module developer, this method may look confusingly similar to `record_user_external_id`, in that both "store a mapping between a user ID and a third-party identifier".

It may help to update the docstring of `record_user_external_id` to clarify exactly what system (authentication through third-party Single Sign-on providers) that it's interacting with.